### PR TITLE
Support for packaging DLCs

### PR DESCRIPTION
**Pull request by Tryum**

### CODE REVIEW

File: nimp/base_commands/package.py [hunk 1]
```diff
@@ -1005,11 +1005,23 @@ def configure_packaging_for_ps5_dlc(env, package_configuration):
             with open(param_file, 'w+') as f:
                 json.dump(params, f, indent=4)
 
+    @staticmethod
+    def configure_packaging_for_xsx_dlc(env, package_configuration):
+        src_microsoft_game_config = env.format('{uproject_dir}/Platforms/XSX/Build/Variants/' + env.variant + '/MicrosoftGame.config')
+        if not os.path.exists(src_microsoft_game_config):
+            raise FileNotFoundError(f'Missing file: {src_microsoft_game_config}, handcrafted MicrosoftGame.config for the DLC is required to exist.')
+        dst_microsoft_game_config = env.format('{uproject_dir}/Saved/StagedBuilds/XSX/Manifest/{unreal_config}/MicrosoftGame.config')
+        logging.info(f'Copying {src_microsoft_game_config} to {dst_microsoft_game_config}')
```
[the remark this answers is not in the document]
Comment (Tryum) [reply nimp/base_commands/package.py]: Fixed at ff091142ed35e7a144948dcc26bf2c0b16153cf2

File: nimp/base_commands/package.py [hunk 2]
```diff
@@ -1013,6 +1013,7 @@ def configure_packaging_for_xsx_dlc(env, package_configuration):
         dst_microsoft_game_config = env.format('{uproject_dir}/Saved/StagedBuilds/XSX/Manifest/{unreal_config}/MicrosoftGame.config')
         logging.info(f'Copying %s to %s', src_microsoft_game_config, dst_microsoft_game_config)
         shutil.copyfile(src_microsoft_game_config, dst_microsoft_game_config)
+        package_configuration.extra_option.append('-DNEDlc')
```
Comment (tdesveaux) [inline nimp/base_commands/package.py:1016]: Sorry might have been not clear.
I meant, do not hardcode this but provide it through the cli arg `--extra-options`

File: nimp/base_commands/package.py [hunk 3]
```diff
@@ -1005,11 +1005,24 @@ def configure_packaging_for_ps5_dlc(env, package_configuration):
...
 
+    @staticmethod
+    def configure_packaging_for_xsx_dlc(env, package_configuration):
+        src_microsoft_game_config = env.format('{uproject_dir}/Platforms/XSX/Build/Variants/' + env.variant + '/MicrosoftGame.config')
+        if not os.path.exists(src_microsoft_game_config):
+            raise FileNotFoundError(f'Missing file: {src_microsoft_game_config}, handcrafted MicrosoftGame.config for the DLC is required to exist.')
+        dst_microsoft_game_config = env.format('{uproject_dir}/Saved/StagedBuilds/XSX/Manifest/{unreal_config}/MicrosoftGame.config')
+        logging.info(f'Copying %s to %s', src_microsoft_game_config, dst_microsoft_game_config)
+        shutil.copyfile(src_microsoft_game_config, dst_microsoft_game_config)
+        package_configuration.extra_options.append('-NoGameOs')
```
Comment (tdesveaux) [inline nimp/base_commands/package.py:1016]: Sorry, I'm annoying with this but I'd like a comment mentioning that it's not from vanilla but a UAT patch of us that ignore remove the `/gameos` arg

**Review by tdesveaux — APPROVED** [review]

File: nimp/base_commands/package.py (hunk 3 above)
Comment (Tryum) [reply nimp/base_commands/package.py:1017]: Is it ok like this ?

Comment (tdesveaux) [reply nimp/base_commands/package.py:1017]: Perfect, thanks!

**Pull request merged**